activation/handler_v2.go

```diff
@@ -757,47 +732,31 @@ func (h *HandlerV2) storeAtx(
 		}
 		return nil
 	}); err != nil {
-		return nil, fmt.Errorf("store atx: %w", err)
+		return fmt.Errorf("store atx: %w", err)
 	}
 
 	atxs.AtxAdded(h.cdb, atx)
 
-	var allMalicious map[types.NodeID]struct{}
-	if malicious || proof != nil {
-		// Combine IDs from the present equivocation set for atx.SmesherID and IDs in atx.Marriages.
-		allMalicious = make(map[types.NodeID]struct{})
-
-		set, err := identities.EquivocationSet(h.cdb, atx.SmesherID)
+	var malicious bool
+	err := h.cdb.WithTx(ctx, func(tx *sql.Tx) error {
+		// malfeasance check happens after storing the ATX because storing updates the marriage set
+		// that is needed for the malfeasance proof
```

Comment:
It shouldn't store **own** (published) ATX which would make itself malicious. This could be resolved by first validating the built ATX in the ATX Builder as we discussed offline. The ATX handler wouldn't re-validate own ATXs.

Reply:
next :)

Reply:
added todo